partitioner/src/main/java/io/confluent/connect/storage/partitioner/TimeBasedPartitioner.java

```diff
@@ -65,8 +65,8 @@ protected void init(
     delim = (String) config.get(StorageCommonConfig.DIRECTORY_DELIM_CONFIG);
     this.partitionDurationMs = partitionDurationMs;
     this.pathFormat = pathFormat;
-    this.formatter = getDateTimeFormatter(pathFormat, timeZone).withLocale(locale);
     try {
+      this.formatter = getDateTimeFormatter(pathFormat, timeZone).withLocale(locale);
```

Comment:
If `pathFormat` is not a proper time format, i.e. `"blah"`, `getMillis()` in the extractor will throw a `IllegalArgumentException`. So, moving this into the try block.